### PR TITLE
perf: Optimize messenger initialization

### LINKED ISSUE #21456: Message history loading takes up to 20 sec
# Bug Report
Follow up of https://github.com/status-im/status-mobile/pull/21411
In this 👆 PR perfomance issue with message loading was already improved but seems that there is a way to make it more faster.

### Reproduction

1. Login to app as a Receiver (a member of Status community)
2. Login to another device as a Sender (a member of Status community)
3. As a Receiver do a logout and close the app.
4. As a Sender wait till Receiver status become offline and send 20+ messages to 1-1 chat with a Receiver.
5. As a Receiver login and open chat with a Sender.

#### Expected behavior
Message history loading takes less then 20 sec

#### Actual behavior
Message history loading takes up to 20 sec

[Receiver_1.zip](https://github.com/user-attachments/files/17434926/Receiver_1.zip)
[Sender_1.zip](https://github.com/user-attachments/files/17434927/Sender_1.zip)

### Additional Information

- Status version: nightly 18.10
- Operating System: Android, iOS

[comment]: # (Please, add logs/notes if necessary)

### PR DESCRIPTION
**Pull request by ilmotta**
Fixes https://github.com/status-im/status-mobile/issues/21456

### Summary

This PR integrates the optimizations coming from the status-go PR https://github.com/status-im/status-go/pull/6106. Now messenger filters are set-up concurrently and we can call the endpoint `wakuext_chatsPreview` twice: once for non-community chats and once for community chats. This way, the user should be able to see some data render before the slow part ends (community-related data). All the hard work is in status-go and executed by @qfrank 

#### Areas that may be impacted

- Login
- Loading of data after login (contacts, chats, etc)

### Steps to test

We need to check two cases for regression (functional and non-functional):

1. You login with a profile with many contacts, chats, opened/joined communities, notifications, etc. If you see all the data after login there's no functional regression.

2. Login with a profile with loads of data and roughly compare the time to see data arrive at different screens (contacts, opened/joined communities, etc) compared to `develop`. In theory, non-community data (e.g. 1:1 chats) can now arrive earlier and doesn't need to wait for the longer times to set-up filters related to community chats. Whichever call to status-go returns first will immediately trigger the client code to render the data.

I already superficially tested both cases, but more extensive QA would be good.

status: ready

### CODE REVIEW

**Review by smohamedjavid — APPROVED** [review]
A much needed optimisation 🚀 
Thanks @qfrank @ilmotta 🫡

**Comment by churik** [comment]
@ilmotta is it ready to be tested?

**Comment by ilmotta** [comment]
> @ilmotta is it ready to be tested?

Yes @churik, PR is ready to be tested. Branch rebased as well. Thanks

**Comment by VolodLytvynenko** [comment]
The current PR is waiting for a rebase of the Status Go [PR](https://github.com/status-im/status-go/pull/6106#issuecomment-2501063256).

File: src/legacy/status_im/data_store/chats.cljs [hunk 1]
```diff
@@ -134,9 +134,9 @@
       unmarshal-members))
 
 (re-frame/reg-fx :fetch-chats-preview
- (fn [{:keys [on-success]}]
+ (fn [{:keys [on-success chat-preview-type]}]
    (json-rpc/call {:method      "wakuext_chatsPreview"
-                   :params      []
+                   :params      [chat-preview-type]
```
Comment (ulisesmac) [inline src/legacy/status_im/data_store/chats.cljs:139]: What are our chat preview types?

**Review by ulisesmac — APPROVED** [review]
Looks good, thanks for the PR

**Comment by mariia-skrypnyk** [comment]
Hey @ilmotta !

Thanks for your PR 🙌

I walk through all steps you've mentioned.

**1.** I returned to https://github.com/status-im/status-mobile/issues/21456 and checked messages loading. As for me they were loaded as it was before +- 20 sec. 
BUT the good thing is I saw all data received 🎉

**2.** According to communities I've compared some Status community channels and found such result:

**Develop vs PR**

Screenshot 2024-11-27 at 17 10 24
Screenshot 2024-11-27 at 17 13 44

As you can see now we have all (!!!) 🎉 messages loaded in channels that are still empty on develop. 

I see no regression in my tests.
E2E went well.

I believe my testing may not fully capture the performance improvements in certain tests, so we should continue monitoring the results in future.

PR can be merged!
Thanks @ilmotta for your relentless efforts to improve the app!

**Comment by ilmotta** [comment]
Testing and proving the improvements in this PR is not easy indeed! Let me just send a :heart:  to @qfrank who managed to pull off the status-go PR.

I'll rebase and merge then. Thank you @mariia-skrypnyk 

Edit: @qfrank I prefer if you merge your PR in status-go and tag it with a new version. Feel free to update this PR with the new status-go tag and merge it if I'm still sleeping 😴, otherwise I'll do this in my morning tomorrow if status-go PR is merged.

**Pull request merged**